src/main/java/UseCases/OtherAccountCases.java

```diff
@@ -1,19 +1,93 @@
+        try {
+            String[] userInfo = new String[6];
+            UserGraphReadWriter reader = new UserGraphReadWriter();
+            UserGraph graph = reader.readFromFile("userGraph.ser");
+            User userclass = graph.getUserByString(userid);
+
+            String displayName = userclass.getDisplayName().getData();
+            String location = userclass.getLocation().getData();
+            String bio = userclass.getBio().getData();
+
+            userInfo[0] = displayName;
+            userInfo[1] = location;
+            userInfo[2] = bio;
+
+            String courseString = "";
+            Courses courses = userclass.getCourses();
+            if (!courses.hidden) {
+                ArrayList<Course> courseList = courses.data;
+                for (int i = 0; i < courseList.size(); i++) {
+                    Course course = courseList.get(i);
+                    if (!course.hidden) {
+                        if (courseString.equals("")) {
+                            courseString += ", " + course.data;
+                        }
+                        else {
+                            courseString += course.data;
+                        }
+                    }
+                }
+            }
 
-        UserGraphReadWriter reader = new UserGraphReadWriter();
-        UserGraph graph = reader.readFromFile("dog.png");
-        User userclass = graph.getUserByString(userid);
-        String[] userInfo = {userclass.getDisplayName().getData(),
-                userclass.getLocation().getData(),
-                userclass.getBio().getData()};
-        return userInfo;
+            String interstsString = "";
+            Interests intersts = userclass.getInterests();
+            if (!intersts.hidden) {
+                HashMap<Integer, Boolean> map = intersts.data;
+                for (int i = 0; i <= 50; i++) {
+                    if (!map.get(i)) {
+                        if (interstsString.equals("")) {
+                            interstsString += ", " + intersts.getInterestAt(i);
+                        }
+                        else {
+                            interstsString +=  intersts.getInterestAt(i);
+                        }
+                    }
+                }
+            }
+
+            Attributes attributes = userclass.getAttributes();
+            String attributesString = "";
+            if (!attributes.hidden) {
+                HashMap<Integer, ArrayList<Object>> map = attributes.data;
+                for (int i = 0; i <= 13; i++) {
+                    ArrayList<Object> atr = map.get(i);
+                    if (!(boolean) atr.get(1)) {
+                        if (attributesString.equals("")) {
+                            attributesString += attributes.getTypeAt(i) + ": " + atr.get(0);
+                        }
+                        else {
+                            attributesString += ", " + attributes.getTypeAt(i) + ": " + atr.get(0);
+                        }
```

Comment:
Not sure what the goal of this section is, but the attributes.getTypeAt() method will return the string corresponding to the type of attribute (like "Eye Color", "Height", etc), not the value like "brown" or "5'10"". If this was what you intended that's fine but just want to make sure you are displaying the correct string here. (I added a getValueAt(int, int) method too if that was what you wanted)